cmd/docker/docker.go

```diff
@@ -50,6 +48,11 @@ func newDockerCommand(dockerCli *command.DockerCli) *cobra.Command {
 	flags.StringVar(&opts.ConfigDir, "config", cliconfig.ConfigDir(), "Location of client config files")
 	opts.Common.InstallFlags(flags)
 
+	// flags must be the top-level command flags, not cmd.Flags()
+	opts.Common.SetDefaultOptions(flags)
+	dockerCli.Initialize(opts)
+	dockerCli.Ping()
```

Comment:
I think this presents a serious UX problem. 

If I run `docker` or `docker --help` to find out how to connect to an alternative host I'm going to get a connection error instead of the help text.

We can't make any API calls before flag parsing is complete, because we won't know which host we're connecting to until after flag parsing. But flag parsing has to happen **after** we setup the child commands. Which means the experimental support also has this issue. I don't think we can ship 1.13 this way.

The order of events needs to be this:

1. setup child commands and flags
2. parse flags
3. initialize client
4. execute command

That means that flags and child commands can not depend on the server version.

I think we should do this:

1. do not try and ping the server
2. go back to the original plan with the version header, where we make a request first then fall back on error.
3. for experimental use `Hidden: isExperimental` on the experimental commands. Add an environment variable `DOCKER_EXPERIMENTAL` that enables showing the commands by setting isExperimental=true`.